app/models/spree/order_contents_decorator.rb

```diff
@@ -56,7 +57,7 @@ def flexi_variants(variant, line_item, product_customizations_values, ad_hoc_opt
         ad_hoc_options_offset_price = line_item.add_ad_hoc_option_values(ad_hoc_option_value_ids)
       end
 
-      line_item.price = variant.price_in(order.currency).amount + customizations_offset_price + ad_hoc_options_offset_price
+      line_item.price = (variant.price_for(line_item.pricing_options) + customizations_offset_price + ad_hoc_options_offset_price).to_d
```

Comment:
What's `line_item.pricing_options` return?

Reply:
`Spree::Variant::PricingOptions.new`

It contains `currency` + `country_iso`